src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs

```diff
@@ -0,0 +1,1642 @@
+                {
+                    N(SyntaxKind.FileKeyword);
+                    N(typeKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Theory]
+        [InlineData(SyntaxKind.ClassKeyword)]
+        [InlineData(SyntaxKind.StructKeyword)]
+        [InlineData(SyntaxKind.InterfaceKeyword)]
+        [InlineData(SyntaxKind.RecordKeyword)]
+        public void FileModifier_02(SyntaxKind typeKeyword)
+        {
+            UsingNode($$"""
+                file partial {{SyntaxFacts.GetText(typeKeyword)}} C { }
+                """);
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxFacts.GetBaseTypeDeclarationKind(typeKeyword));
+                {
+                    N(SyntaxKind.FileKeyword);
+                    N(SyntaxKind.PartialKeyword);
+                    N(typeKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void FileModifier_02_Enum()
+        {
+            UsingNode($$"""
+                file partial enum C { }
+                """,
+                // (1,6): error CS0267: The 'partial' modifier can only appear immediately before 'class', 'record', 'struct', 'interface', or a method return type.
+                // file partial enum C { }
+                Diagnostic(ErrorCode.ERR_PartialMisplaced, "partial").WithLocation(1, 6));
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.EnumDeclaration);
+                {
+                    N(SyntaxKind.FileKeyword);
+                    N(SyntaxKind.PartialKeyword);
+                    N(SyntaxKind.EnumKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Theory]
+        [InlineData(SyntaxKind.ClassKeyword)]
+        [InlineData(SyntaxKind.StructKeyword)]
+        [InlineData(SyntaxKind.InterfaceKeyword)]
+        public void FileModifier_03(SyntaxKind typeKeyword)
+        {
+            UsingNode($$"""
+                partial file {{SyntaxFacts.GetText(typeKeyword)}} C { }
+                """,
+                // (1,14): error CS1002: ; expected
+                // partial file {{SyntaxFacts.GetText(typeKeyword)}} C { }
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, SyntaxFacts.GetText(typeKeyword)).WithLocation(1, 14));
```

Comment:
After the change to `IsPossibleTypeDeclarationStart`, I expect this error to change (for the better).

Reply:
Followed up on this in https://github.com/dotnet/roslyn/pull/60823#discussion_r860311430.